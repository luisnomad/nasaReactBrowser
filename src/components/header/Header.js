import React from "react";
import { Link } from "react-router-dom";
import styles from './Header.scss';
import classNames from 'classnames/bind';

const cx = classNames.bind(styles)

// The Header creates links that can be used to navigate
// between routes.
const Header = () => (
    <header className={cx('header')}>
        <nav>
            <ul>
                <li>
                    <Link to="/">Home</Link>
                </li>
                <li>
                    <Link to="/search">Search</Link>
                </li>
            </ul>
        </nav>
    </header>
);

export default Header;
