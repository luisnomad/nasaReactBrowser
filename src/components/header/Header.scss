@import '../../styles/variables';

.header {
  background-color: $nasa_blue;
  color: $nasa_red;

  a {
    text-decoration: none;
    &:visited, &:active {
      color: $nasa_red;
    }
  }
}

.navBar {
  list-style: none;
  margin: 0;
  padding: 0.5rem;

  li {
    display: inline;
    margin-right: 1rem;
  }
}