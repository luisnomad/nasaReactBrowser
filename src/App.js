import React, { Fragment } from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from './components/header';
import Home from './components/home';
import { Search, Asset } from './containers';

import styles from './App.scss';
import classNames from 'classnames/bind';

const cx = classNames.bind(styles);

const App = () => (
    <Fragment>
        <main role="main">
            <Switch>
                <Route exact path='/' component={Home}/>
                <Route exact path='/search' component={Search}/>
                <Route path='/asset/:asset' component={Asset}/>
            </Switch>
        </main>
    </Fragment>
);

export default App;