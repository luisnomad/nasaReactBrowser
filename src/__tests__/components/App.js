import React from 'react';
import { shallow } from 'enzyme';
import App from '../../App';
import Header from '../../components/header';

describe('<App />', () => {

    const wrapped = shallow (<App />);

    it('Should render a Header', () => {
        expect(wrapped.find(Header).length).toEqual(1);
    });
});