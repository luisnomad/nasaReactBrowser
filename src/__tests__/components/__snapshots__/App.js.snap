// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render correctly 1`] = `
<Fragment>
  <main
    role="main"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/search"
      />
      <Route
        component={[Function]}
        path="/asset/:asset"
      />
    </Switch>
  </main>
</Fragment>
`;
